crates/service-protocol/src/message/encoding.rs

```diff
@@ -116,21 +124,26 @@ fn encode_msg(msg: &ProtocolMessage, buf: &mut impl BufMut) -> Result<(), prost:
 
 /// Stateful decoder to decode [`ProtocolMessage`]
 pub struct Decoder {
+    _service_protocol_version: ServiceProtocolVersion,
     buf: SegmentedBuf<Bytes>,
     state: DecoderState,
     message_size_warning: usize,
     message_size_limit: usize,
 }
 
-impl Default for Decoder {
-    fn default() -> Self {
-        Decoder::new(usize::MAX, None)
-    }
-}
-
 impl Decoder {
-    pub fn new(message_size_warning: usize, message_size_limit: Option<usize>) -> Self {
+    pub fn new(
+        service_protocol_version: ServiceProtocolVersion,
+        message_size_warning: usize,
+        message_size_limit: Option<usize>,
+    ) -> Self {
+        assert_eq!(
+            service_protocol_version,
+            ServiceProtocolVersion::V1,
+            "Decoder only supports service protocol version V1"
+        );
         Self {
+            _service_protocol_version: service_protocol_version,
```

Comment:
No need to store this in the Decoder. Same for Encoder.

Reply:
This is indeed not used right now and left there for the future. I can remove it.